<!--
  2 failure:  :disappointed_rel..., :microscope: Ther...
  0 warning: 
  1 messages
  
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  :disappointed_relieved: `npm test` failed. See below for details.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  :microscope: There are library changes, but no changes to the unit tests. That's OK as long as you're refactoring existing code, but will require an admin to merge this PR. Please see <a href="https://github.com/janvennemann/titanium_mobile/blob/gc-fix-v2-7_4_X/README.md#unit-tests">README.md#unit-tests</a> for docs on unit testing.
  </td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  ```
> titanium-mobile@7.4.1 test /Users/build/jenkins/workspace/dk_titanium_mobile_PR-10370-EPPXGKSZ573XRVMSMAAIVFEM4D42MGQGZFV623R35XUNXR64E4MA
> grunt

Running "appcJs:src:lintOnly" (appcJs) task

Running "eslint:src" (eslint) task

Running "checkFormat:ios" (checkFormat) task
Fatal error: Error: Formatting incorrect on "iphone/Classes/KrollObject.m", proposed changes: <?xml version='1.0'?>
<replacements xml:space='preserve' incomplete_format='false'>
<replacement offset='13191' length='10'>&#10;&#10;    </replacement>
<replacement offset='47783' length='6'>&#10;&#10;  </replacement>
<replacement offset='47869' length='4'>&#10;&#10;</replacement>
<replacement offset='48163' length='6'>&#10;&#10;  </replacement>
<replacement offset='48644' length='6'>&#10;&#10;  </replacement>
<replacement offset='48730' length='4'>&#10;&#10;</replacement>
<replacement offset='49031' length='6'>&#10;&#10;  </replacement>
</replacements>

npm ERR! Test failed.  See above for more details.

```
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

--- a/iphone/Classes/KrollObject.m
+++ b/iphone/Classes/KrollObject.m
@@ -424,7 +424,7 @@ bool KrollHasInstance(TiContextRef ctx, TiObjectRef constructor, TiValueRef poss
     bridge = (KrollBridge *)[context_ delegate];
     jsobject = TiObjectMake(jsContext, [[self class] jsClassRef], self);
     targetable = [target conformsToProtocol:@protocol(KrollTargetable)];
-    
+
     self.gcSafeguarded = NO;
   }
   return self;
@@ -949,14 +949,14 @@ bool KrollHasInstance(TiContextRef ctx, TiObjectRef constructor, TiValueRef poss
 TI_INLINE TiStringRef TiStringCreateWithPointerValue(int value)
 {
   /*
-	 *	When we note proxies, we need to come up with a property name
-	 *	that is unique. We previously did an nsstring with format
-	 *	of __PX%X, but this method is so often called, and allocating a string
-	 *	can be a waste, so it's better to jump straight to something hardwired
-	 *
-	 *	No sense in doing hex when so many more characters are valid property
-	 *	characters. So we do it in chunks of 6 bits, from '<' (60) to '{' (123)
-	 */
+   *	When we note proxies, we need to come up with a property name
+   *	that is unique. We previously did an nsstring with format
+   *	of __PX%X, but this method is so often called, and allocating a string
+   *	can be a waste, so it's better to jump straight to something hardwired
+   *
+   *	No sense in doing hex when so many more characters are valid property
+   *	characters. So we do it in chunks of 6 bits, from '<' (60) to '{' (123)
+   */
   char result[10];
   result[0] = '_';
   result[1] = '_';
@@ -1382,7 +1382,7 @@ TiThreadPerformOnMainThread(mainBlock, NO);
 #ifdef USE_JSCORE_FRAMEWORK
 /**
  Protects the underlying JSObjectRef from being accidentally GC'ed.
- 
+
  Upon proxy creation there is a small timeframe between creating the KrollObject
  with its JSObject and the JSObject actually getting referenced in the JS object graph.
  If JSC's garbage collection happens during this time the JSObject is lost and eventually
@@ -1393,11 +1393,11 @@ TiThreadPerformOnMainThread(mainBlock, NO);
   if (self.isGcSafeguarded == YES) {
     return;
   }
-  
+
   if (finalized == YES || jsContext == NULL || jsobject == NULL) {
     return;
   }
-  
+
 #ifdef TI_USE_KROLL_THREAD
   if (![context isKJSThread]) {
     NSOperation *safeProtect = [[NSInvocationOperation alloc] initWithTarget:self selector:@selector(applyGarbageCollectionSafeguard) object:nil];
@@ -1406,14 +1406,14 @@ TiThreadPerformOnMainThread(mainBlock, NO);
     return;
   }
 #endif
-  
+
   TiValueProtect(jsContext, jsobject);
   self.gcSafeguarded = YES;
 }
 
 /**
  Removes the garbage collection safeguard by unprotecting the JSObjectRef again.
- 
+
  This must only be called immediately before the JSObjectRef gets inserted into the JS object
  graph. Even better would be after the insertion but there is currently no viable place to do
  that in our proxy creation flow.
@@ -1423,11 +1423,11 @@ TiThreadPerformOnMainThread(mainBlock, NO);
   if (self.isGcSafeguarded == NO) {
     return;
   }
-  
+
   if (finalized == YES || jsContext == NULL || jsobject == NULL) {
     return;
   }
-  
+
 #ifdef TI_USE_KROLL_THREAD
   if (![context isKJSThread]) {
     NSOperation *safeUnprotect = [[NSInvocationOperation alloc] initWithTarget:self selector:@selector(removeGarbageCollectionSafeguard) object:nil];
@@ -1436,7 +1436,7 @@ TiThreadPerformOnMainThread(mainBlock, NO);
     return;
   }
 #endif
-  
+
   TiValueUnprotect(jsContext, jsobject);
   self.gcSafeguarded = NO;
 }

